public/app/plugins/datasource/prometheus/querybuilder/shared/OperationParamEditor.tsx

```diff
@@ -45,16 +45,22 @@ function SelectInputParamEditor({
   operationIndex,
   onChange,
 }: QueryBuilderOperationParamEditorProps) {
-  const selectOptions = paramDef.options!.map((option) => ({
-    label: option as string,
-    value: option as string,
-  }));
+  let selectOptions = paramDef.options as Array<SelectableValue<any>>;
+
+  if (!selectOptions[0]?.label) {
+    selectOptions = paramDef.options!.map((option) => ({
+      label: option as string,
```

Comment:
nit: Should this be `label: option.toString()` to make sure that we have correct type as `paramDef.options` can be also array of numbers?  And also the same below.